packages/create-plugin/src/utils/utils.templates.ts

```diff
@@ -88,23 +92,70 @@ export function renderTemplateFromFile(templateFile: string, data?: any) {
   return renderHandlebarsTemplate(fs.readFileSync(templateFile).toString(), data);
 }
 
-export function getTemplateData(): TemplateData {
+export function buildProxyFromJson() {
   const pluginJson = getPluginJson();
+  return new Proxy(pluginJson, {
```

Comment:
What's the reason for using a `Proxy` here? (Instead of using a simple object with overrides?)

Reply:
I have used Proxy, because we have two different types of objects (CliArgs and json) which we want to use as an argument of getTemplateData. And each object has different names for the same property, for example at CliArgs we have `pluginId` and in json we have just an `id`. We can update objects to have the same naming although the feature scope would explode
In case of overrides the original object will be modified and it will be harder to understand what is going on in the future. The mapping will not be explicit. In case of Proxy we can see which properties are there
If we decide to unify naming at some point, we will do so, delete Proxy and rest of the code will stay the same

Reply:
I think we could just static-type the input of `getTemplateData()` and the callsite could take care of passing in data in the right format (and doing the data-mapping necessary). I would personally find that simpler to scan, and also the "knowledge" of call-site specific naming (CliArgs and json) would stay closer to where they are created.

Reply:
I think I agree with Levi on this one. The proxy usage seems a bit much to me compared to a function that maybe takes in CliArgs else gets the pluginJson file then does the work to "evolve" the information we need into the object thats expected to be passed into `getTemplateData`.